File: java/src/main/java/cn/leetcode/array/$413_ArithmeticSlices.java
package cn.leetcode.array;

public class $413_ArithmeticSlices {
    public int numberOfArithmeticSlices(int[] nums) {
        if(nums.length < 3){
            return 0;
        }
        int ans = 0;
        int t = 0;
        int d = nums[1] - nums[0];
        for (int i = 2; i < nums.length; i++) {
            if(nums[i] - nums[i-1] == d){
                t++;
            }else{
                d = nums[i] - nums[i-1];
                t = 0;
            }
            ans += t;
        }
        return ans;
    }

    public static void main(String[] args) {
        int ans = new $413_ArithmeticSlices().numberOfArithmeticSlices(new int[]{1,2,3,8,9,10});
        System.out.println(ans);
    }
}

File: python/array/$413_ArithmeticSlices.py
from typing import List


class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return 0
        ans, t = 0, 0
        d = nums[1] - nums[0]
        for i in range(2, len(nums)):
            if nums[i] - nums[i-1] == d:
                t += 1
            else:
                t = 0
                d = nums[i] - nums[i-1]
            ans += t
        return ans


res = Solution().numberOfArithmeticSlices([2,1,3,4,2,3])
print()
